api/src/main/java/org/apache/iceberg/IncrTableScan.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+public interface IncrTableScan extends TableScan {
+  IncrTableScan readRange(long startSnapshotId, long endSnapshotId);
```

Comment:
It seems strange to me that this is optional in an incremental scan. What is the default behavior if this isn't called?

Reply:
This shouldn't be optional. If API is reflecting that, then maybe, an alternative could be that we add a method to Table scan like u suggested in the mailing list - `TableScan#appendsBetween(s1, s2)`

Reply:
I like the `appendsBetween` idea. That helps keep the `Table` API smaller and could reuse a lot of the existing scan planning code.

Reply:
I've keep the name as `incrementalScan` but the idea is incorportaed